### PR TITLE
 #820 - Make sure AccumuloClient is closed

### PR DESCRIPTION
**Pull request by mikewalch**
* Closed AccumuloClient in many places where it was not being closed
* Updated method names to reflect if AccumuloClient should be closed.
* 'create' in method name means returned AccumuloClient must closed by user
* 'get' in method name means returned AccumuloClient doesn't need to be
  closed by user

### CODE REVIEW

File: test/src/main/java/org/apache/accumulo/test/AuditMessageIT.java [hunk 1]
```diff
@@ -207,7 +207,8 @@ public void testTableOperationsAudits() throws AccumuloException, AccumuloSecuri
 
     // Connect as Audit User and do a bunch of stuff.
     // Testing activity begins here
-    auditAccumuloClient = getCluster().getAccumuloClient(AUDIT_USER_1, new PasswordToken(PASSWORD));
+    auditAccumuloClient = getCluster().createAccumuloClient(AUDIT_USER_1,
```
Comment (keith-turner) [inline test/src/main/java/org/apache/accumulo/test/AuditMessageIT.java]: why not close this?

File: test/src/main/java/org/apache/accumulo/test/CompactionRateLimitingIT.java [hunk 2]
```diff
@@ -47,7 +47,7 @@ public void configure(MiniAccumuloConfigImpl cfg, Configuration fsConf) {
   public void majorCompactionsAreRateLimited() throws Exception {
     long bytesWritten = 0;
     String tableName = getUniqueNames(1)[0];
-    AccumuloClient client = getCluster().getAccumuloClient("root",
+    AccumuloClient client = getCluster().createAccumuloClient("root",
```
Comment (keith-turner) [inline test/src/main/java/org/apache/accumulo/test/CompactionRateLimitingIT.java]: Why not close this?

File: test/src/main/java/org/apache/accumulo/test/ExistingMacIT.java [hunk 3]
```diff
@@ -87,7 +87,7 @@ private void createEmptyConfig(File confFile) throws IOException {
   @Test
   public void testExistingInstance() throws Exception {
 
-    AccumuloClient client = getCluster().getAccumuloClient("root",
+    AccumuloClient client = getCluster().createAccumuloClient("root",
```
Comment (keith-turner) [inline test/src/main/java/org/apache/accumulo/test/ExistingMacIT.java]: close this?

File: test/src/main/java/org/apache/accumulo/test/ShellServerIT.java [hunk 4]
```diff
@@ -304,7 +304,7 @@ public static void setupMiniCluster() throws Exception {
 
     traceProcess = getCluster().exec(TraceServer.class);
 
-    AccumuloClient client = getCluster().getAccumuloClient(getPrincipal(), getToken());
+    AccumuloClient client = getCluster().createAccumuloClient(getPrincipal(), getToken());
```
Comment (keith-turner) [inline test/src/main/java/org/apache/accumulo/test/ShellServerIT.java]: close?

File: test/src/main/java/org/apache/accumulo/test/functional/KerberosIT.java [hunk 5]
```diff
@@ -164,7 +164,7 @@ public void testAdminUser() throws Exception {
     UserGroupInformation ugi = UserGroupInformation.loginUserFromKeytabAndReturnUGI(
         rootUser.getPrincipal(), rootUser.getKeytab().getAbsolutePath());
     ugi.doAs((PrivilegedExceptionAction<Void>) () -> {
-      final AccumuloClient client = mac.getAccumuloClient(rootUser.getPrincipal(),
+      final AccumuloClient client = mac.createAccumuloClient(rootUser.getPrincipal(),
```
Comment (keith-turner) [inline test/src/main/java/org/apache/accumulo/test/functional/KerberosIT.java]: close?

File: test/src/main/java/org/apache/accumulo/test/AuditMessageIT.java (hunk 1 above)
Comment (mikewalch) [reply test/src/main/java/org/apache/accumulo/test/AuditMessageIT.java]: I wanted to limit the size of this PR.  I won't close #820 until everything has been closed or other issues have been created for remaining work

**Pull request merged**